src/bionics_ui.cpp

```diff
@@ -71,37 +71,31 @@ void draw_bionics_titlebar( WINDOW *window, player *p, bionic_menu_mode mode )
     wrefresh( window );
 }
 
-const auto separator = []( std::ostringstream &s )
-{
-    return ( int )( s.tellp() ) != 0 ? ", " : "";
-};
-
 //builds the power usage string of a given bionic
 std::string build_bionic_poweronly_string( bionic const &bio )
 {
-    std::ostringstream power_desc;
-    if( bionic_info( bio.id ).power_over_time > 0 && bionic_info( bio.id ).charge_time > 0 ) {
-        power_desc << (
-                       bionic_info( bio.id ).charge_time == 1
-                       ? string_format( _( "%d PU/turn" ),
-                                        bionic_info( bio.id ).power_over_time )
-                       : string_format( _( "%d PU/%d turns" ),
-                                        bionic_info( bio.id ).power_over_time,
-                                        bionic_info( bio.id ).charge_time ) );
-    }
-    if( bionic_info( bio.id ).power_activate > 0 && !bionic_info( bio.id ).charge_time ) {
-        power_desc << separator( power_desc ) << string_format( _( "%d PU act" ),
-                   bionic_info( bio.id ).power_activate );
-    }
-    if( bionic_info( bio.id ).power_deactivate > 0 && !bionic_info( bio.id ).charge_time ) {
-        power_desc << separator( power_desc ) << string_format( _( "%d PU deact" ),
-                   bionic_info( bio.id ).power_deactivate );
+    const bionic_data &bio_data( bionic_info( bio.id ) );
+    std::vector<std::string> properties;
+
+    if( bio_data.charge_time > 0 ) {
+        if( bio_data.power_over_time > 0 ) {
+            properties.push_back( ( bio_data.charge_time == 1
```

Comment:
Looks like you can remove a pair of brackets here, currently it's `push_back( ( ... ) )`.